tests/bluechi_test/service.py

```diff
@@ -38,37 +46,61 @@ class Option(enum.Enum):
 
 
 class Service():
-    """An empty service, which only contains predefined sections"""
-    def __init__(self, name: str = None) -> None:
-        self.cp = configparser.ConfigParser()
-        # Enable case sensitivity
-        self.cp.optionxform = lambda option: option
-        self.name = name
+    """A base service class, which by default contains only predefined sections"""
+    def __init__(self, name: str = SERVICE_NAME, directory: str = SERVICE_DIRECTORY) -> None:
+        if not name:
+            raise Exception(f"Service name is '{name}', but cannot be empty!")
+
+        if not directory:
+            raise Exception(f"Service directory is '{directory}', but cannot be empty!")
+
+        self._svc_path = pathlib.Path(directory) / pathlib.Path(name)
```

Comment:
```suggestion
        self._svc_path = pathlib.Path(directory).joinPath(pathlib.Path(name))
```
I don't know about any different behavior between those two approaches, though.

Reply:
There shouldn't be any significant difference: https://github.com/python/cpython/blob/main/Lib/pathlib/_abc.py#L395-L407